engine/storage/src/main/java/org/apache/cloudstack/storage/endpoint/DefaultEndPointSelector.java

```diff
@@ -84,7 +85,8 @@ public class DefaultEndPointSelector implements EndPointSelector {
                             + "left join host_details hd on h.id=hd.host_id and hd.name='" + HOST_VOLUME_ENCRYPTION + "' "
                             + "where h.status = 'Up' and h.type = 'Routing' and h.resource_state = 'Enabled' and s.pool_id = ? ";
 
-    private String findOneHypervisorHostInScopeByType = "select h.id from host h where h.status = 'Up' and h.hypervisor_type = ? ";
+    private String findOneHypervisorHostInScopeByHypervisorType = "select h.id from " +
+            "host h where h.status = 'Up' and h.type = 'Routing' and h.resource_state = 'Enabled' and h.hypervisor_type = ? ";
```

Comment:
`h.resource_state = 'Enabled'`

can disabled hosts handle the process ?

Reply:
Ah, good point, thanks